apps/io-wallet-user-func/src/infra/http/handlers/set-wallet-instance-status.ts

```diff
@@ -33,16 +34,32 @@ const requireSetWalletInstanceStatusBody: (
 
 export const SetWalletInstanceStatusHandler = H.of((req: H.HttpRequest) =>
   pipe(
-    sequenceS(RTE.ApplyPar)({
-      body: pipe(req, requireSetWalletInstanceStatusBody, RTE.fromEither),
-      fiscalCode: pipe(req, requireWhitelistedFiscalCodeFromToken),
-      walletInstanceId: pipe(req, requireWalletInstanceId, RTE.fromEither),
-    }),
-    // invoke PID issuer services to revoke all credentials for that user
-    RTE.chainFirstW(({ fiscalCode }) => revokeAllCredentials(fiscalCode)),
-    // access our database to revoke the wallet instance
-    RTE.chainW(({ fiscalCode, walletInstanceId }) =>
-      revokeUserWalletInstances(fiscalCode, [walletInstanceId]),
+    req,
+    requireWhitelistedFiscalCodeFromToken,
+    RTE.chainW((fiscalCode) =>
+      pipe(
+        sequenceS(E.Apply)({
+          body: pipe(req, requireSetWalletInstanceStatusBody),
+          walletInstanceId: pipe(req, requireWalletInstanceId),
+        }),
+        RTE.fromEither,
+        // invoke PID issuer services to revoke all credentials for that user
+        RTE.chainFirstW(() => revokeAllCredentials(fiscalCode)),
+        // access our database to revoke the wallet instance
+        RTE.chainW(({ walletInstanceId }) =>
+          revokeUserWalletInstances(fiscalCode, [walletInstanceId]),
+        ),
+        RTE.orElseFirstW((error) =>
+          pipe(
+            sendTelemetryException(error, {
+              fiscalCode,
+              pathParameter: req.path,
+              payload: req.body,
+            }),
+            RTE.fromReader,
+          ),
+        ),
+      ),
     ),
     RTE.map(() => H.empty),
     RTE.orElseW(logErrorAndReturnResponse),
```

Comment:
I don't know if it's feasible but isn't it better to add the application insight log inside `logErrorAndReturnResponse` ?